chain/client/src/client_actor.rs

```diff
@@ -1481,9 +1481,13 @@ impl ClientActor {
         let head = self.client.chain.head()?;
         let mut is_syncing = self.client.sync_status.is_syncing();
 
-        let peer_info = if let Some(peer_info) =
-            self.network_info.highest_height_peers.choose(&mut thread_rng())
-        {
+        // Only consider peers whose latest block is not invalid blocks
+        let eligible_peers = self
+            .network_info
+            .highest_height_peers
+            .iter()
+            .filter(|p| !self.client.chain.is_block_invalid(&p.highest_block_hash));
```

Comment:
Should we warn! if we did filter someone out, or would that be too spammy? Just thinking of how we would know this happens if we were looking for it. Maybe something on the debug page? But then... what exactly should be on there... It's ok if we can't think of anything for now.

Reply:
Oh yes, great questions. So I think we can 
- show what's in invalid_blocks in debug page
- add a metric to track how many peers is filtered
I'll add that in this PR today

Reply:
Well,

> show what's in invalid_blocks in debug page

Not sure if that's gonna be helpful. Also it's an lru cache of 5000; it's going to potentially flood the UI with many blocks that are arbitrarily old.

> add a metric to track how many peers is filtered

I guess this would be a gauge? If it's a counter it would need two metrics; one for how many times we have reported this.

Reply:
> Not sure if that's gonna be helpful. Also it's an lru cache of 5000; it's going to potentially flood the UI with many blocks that are arbitrarily old.
Hopefully we won't have that many invalid blocks :) I'll put it at the end of the page, so if it does flood, at least it won't affect readability that much 
> I guess this would be a gauge? If it's a counter it would need two metrics; one for how many times we have reported this.
yeah it will be a gauge

Reply:
And to your original question, yeah I think warn might spam the log. The sync check is triggered every 10ms.